freqtrade/strategy/strategyupdater.py

```diff
@@ -0,0 +1,253 @@
+
+        # space is not yet transferred from buy/sell to entry/exit and thereby has to be skipped.
+        if isinstance(node, ast_comments.keyword):
+            if node.arg == "space":
+                return node
+
+        # from here on this is the original function.
+        for field, old_value in ast_comments.iter_fields(node):
+            if isinstance(old_value, list):
+                new_values = []
+                for value in old_value:
+                    if isinstance(value, ast_comments.AST):
+                        value = self.visit(value)
+                        if value is None:
+                            continue
+                        elif not isinstance(value, ast_comments.AST):
+                            new_values.extend(value)
+                            continue
+                    new_values.append(value)
+                old_value[:] = new_values
+            elif isinstance(old_value, ast_comments.AST):
+                new_node = self.visit(old_value)
+                if new_node is None:
+                    delattr(node, field)
+                else:
+                    setattr(node, field, new_node)
+        return node
+
+    def visit_Expr(self, node):
+        if hasattr(node.value, "left") and hasattr(node.value.left, "id"):
+            node.value.left.id = self.check_dict(StrategyUpdater.name_mapping, node.value.left.id)
+            self.visit(node.value)
+        return node
+
+    # Renames an element if contained inside a dictionary.
+    @staticmethod
+    def check_dict(current_dict: dict, element: str):
+        if element in current_dict:
+            element = current_dict[element]
+        return element
+
+    def visit_arguments(self, node):
+        if isinstance(node.args, list):
+            for arg in node.args:
+                arg.arg = self.check_dict(StrategyUpdater.name_mapping, arg.arg)
+        return node
+
+    def visit_Name(self, node):
+        # if the name is in the mapping, update it
+        node.id = self.check_dict(StrategyUpdater.name_mapping, node.id)
+        return node
+
+    def visit_Import(self, node):
+        # do not update the names in import statements
+        return node
+
+    def visit_ImportFrom(self, node):
+        # if hasattr(node, "module"):
+        #    if node.module == "freqtrade.strategy.hyper":
+        #        node.module = "freqtrade.strategy"
+        return node
+
+    def visit_If(self, node: ast_comments.If):
+        for child in ast_comments.iter_child_nodes(node):
+            self.visit(child)
+        return node
+
+    def visit_FunctionDef(self, node):
+        node.name = self.check_dict(StrategyUpdater.function_mapping, node.name)
+        self.generic_visit(node)
+        return node
+
+    def visit_Attribute(self, node):
+        if (
+                isinstance(node.value, ast_comments.Name)
+                and node.value.id == 'trades'
+                and node.attr == 'nr_of_successful_buys'
```

Comment:
Am i reading this right that this is explicitly updating `trades.nr_successful_buys` to `trades.nr_successful_entries`?

I think the more common occurance for this will be `trade.nr_successful_buys` (singular - as is the name for the `trade` parameter) - which would currently fail.